### ISSUE TITLE
List branding/graphics used for Omni Core

### ISSUE THREAD

**Issue by dexX7** [OmniLayer/omnicore#214]
It was requested to provide help to "white label" Omni Core, and one low hanging solution could be to switch the splash screen image and friends.

We may document the files used in this context, such as:

- https://github.com/OmniLayer/omnicore/blob/omnicore-0.0.10/src/qt/res/images/splash.png
- https://github.com/OmniLayer/omnicore/blob/omnicore-0.0.10/src/qt/res/images/splash_testnet.png
- https://github.com/OmniLayer/omnicore/blob/omnicore-0.0.10/share/pixmaps/nsis-wizard.bmp
- https://github.com/OmniLayer/omnicore/blob/omnicore-0.0.10/share/pixmaps/nsis-header.bmp
- the icons
- ...

This also raises the question how far we should actively support or discourage this.

I feel comfortable with a rebranded version with other graphics, but it should be clear that any alternative version must follow the consensus rules, and we should probably mention this in the documentation (if there should be one).

**Comment by zathras-crypto** [comment]
> This also raises the question how far we should actively support or discourage this.

I'm happy for people to rebrand as they see fit, but this raises the question of release process.

> But it should be clear that any alternative version must follow the consensus rules

That's my main concern, a white-labelled version of Omni Core must follow the same requirements as the reference code.  So when we activate features that require an upgrade, the whitelabelled version requires an upgrade just the same.  Making the assumption that "whitelabellers" will be responsive and issue new releases at the same time as us is risky.  For example lets say there are users using:

* OmniCore reference client 0.0.10 provided by Omni Foundation
* BrandedCore 0.0.10 provided by CompanyA
* OtherBrandedCore 0.0.10 provided by CompanyB

Now we then release a feature that requires an update to 0.0.11.  We release 0.0.11, CompanyA releases BrandedCore 0.0.11 on time, but CompanyB does not provide an updated OtherBrandedCore 0.0.11.

Users of OtherBrandedCore are thus stuck on 0.0.10 because they have not been provided with a newer release.   This shuts down when the new feature goes active and users are unhappy.

Of course in this scenario users of OtherBrandedCore can just switch to the reference Omni Core, but that assumes they *know about* the reference Omni Core which isn't a safe assumption.  

TL:DR; what does this mean for the perception of our project, if we hand off release management to third parties via their own white-labelled binaries?

Hope that makes sense!

**Comment by dexX7** [comment]
> Hope that makes sense!

You addressed the relevant points.

In general I think diversity would be awesome, and the tricky part I see is to keep things synchronized.

If only some images are swapped, then it's rather unlikely there are conflicts, but if the whole UI is modified, an upgrade is probably much more difficult.

I suggest to start simple and list "low impact" style modifications such as images, and depending on what we have, we may group them (IIRC some icons are in `share/`, while others are in `src/qt/...` etc.).

**Comment by zathras-crypto** [comment]
> In general I think diversity would be awesome, and the tricky part I see is to keep things synchronized.

Yeah I'm all for diversity too it's just we are changing things so fast! white-labellers may not keep up with releases leading to a fractured code base among users. 

> If only some images are swapped, then it's rather unlikely there are conflicts

Yep I agree, my concern is more:
* User A runs the reference client "Omni Wallet Desktop" from our github release
* User B runs a white-labelled client "White Wallet Wizard" downloaded from CompanyXYZ
* There is a feature activation that requires an update to 0.11

Now what happens when CompanyXYZ don't make available an update to "White Wallet Wizard" to 0.11?  Now User B has no upgrade path for "White Wallet Wizard" and may not know what "Omni Wallet Desktop" is, or that it's a compatible replacement or where to get it.  Perhaps it's just a communications/PR problem but it's one that comes to mind...

One other issue to consider is that of build hashes, our signatures will only be for the reference builds we supply.  If we start to encourage/support users to download other non-reference clients from third parties are we adding risk?  Eg if we have blanket support for anyone supplying their own whitelabelled binaries with image & label modifications then users can no longer verify those builds, and a bad actor could include other less-benign changes.

Perhaps some kind of "partner" program is viable where builds could be vetted and verified - just thinking out loud...

> I suggest to start simple and list "low impact" style modifications such as images

I'm kind of on the fence about it - technically swapping out an image is an incredibly minor change and of no real concern - it's the non-technical questions like fractured user base, forks from customizations to code & white-labellers falling behind with releases that I'm thinking about when I think of the context of white labelling...

**Comment by dexX7** [comment]
> Yeah I'm all for diversity too it's just we are changing things so fast!

Is that so? Keep in mind: the last public release is still 0.0.9.

I see it this way:

- documentation is always useful, either for us, for new developers, or something else
- if anyone wants this, there are two options: we try to convince the party not to do it, or we provide support and outline the risks

Though you raise important points regarding the build process, and the updating.

**Comment by zathras-crypto** [comment]
> Is that so? Keep in mind: the last public release is still 0.0.9.

Now that is a very good point - it's easy to loose sight of that!

I'm generally very supportive of anything that will grow the ecosystem, I just wanted us to look at all the angles :)
